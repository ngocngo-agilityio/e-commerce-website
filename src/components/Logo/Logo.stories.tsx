import type { Meta, StoryObj } from '@storybook/react';
import { fn } from '@storybook/test';
import Logo from '.';

const meta = {
  title: 'Logo',
  component: Logo,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    backgroundColor: { control: 'color' },
  },
  args: { onClick: fn() },
} satisfies Meta<typeof Logo>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {},
};
