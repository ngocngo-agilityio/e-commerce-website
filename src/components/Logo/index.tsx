// Libs
import { memo } from 'react';
import { Heading } from '@chakra-ui/react';
import Link from 'next/link';

// Constants
import { APP_ROUTERS } from '@constants';

const Logo = (): JSX.Element => {
  return (
    <Heading as="h1" fontFamily="lato" color="logo" fontWeight="normal">
      <Link href={APP_ROUTERS.HOME_PAGE}>NorthStar</Link>
    </Heading>
  );
};

export default memo(Logo);
