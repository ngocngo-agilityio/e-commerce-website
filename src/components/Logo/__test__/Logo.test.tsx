// Libs
import { render, screen } from '@testing-library/react';

// Components
import Logo from '..';

describe('Logo component', () => {
  test('should render Logo successfully', () => {
    render(<Logo />);

    expect(screen.getByText('NorthStar')).toBeInTheDocument();
  });

  test('should match snapshot for Logo', () => {
    const { container } = render(<Logo />);

    expect(container).toMatchSnapshot();
  });
});
