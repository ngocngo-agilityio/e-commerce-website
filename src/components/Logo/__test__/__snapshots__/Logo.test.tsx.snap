// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component should match snapshot for Logo 1`] = `
<div>
  <h1
    class="chakra-heading css-jouhmg"
  >
    <a
      href="/products"
    >
      NorthStar
    </a>
  </h1>
</div>
`;
